.github/workflows/test-and-deploy.yml

```diff
@@ -10,37 +10,64 @@ jobs:
       steps:
         - uses: actions/checkout@v2
         - uses: jurplel/install-qt-action@v2
-        - uses: s-weigand/setup-conda@v1.0.7
-          name: Setup Conda
+        - uses: conda-incubator/setup-miniconda@v2
           with:
+            auto-update-conda: true
             python-version: 3.8
-            conda-channels: anaconda, conda-forge
-            activate-conda: true
+            channels: conda-forge
+            auto-activate-base: true
+        - name: Map the Branch Name to a Conda Environment
+          uses: neutrons/branch-mapper@modifiedMain
+          id: envname
+          with:
+            prefix: "neutron-imaging"
+            suffix-default: "-dev"
         - name: Install dependencies
           timeout-minutes: 40
+          shell: bash -l {0}
           run: |
             python -m pip install --upgrade pip
-            conda env update -n base --file=conda/environment.yml
-            conda install pytest
+            conda env update -n base --file=conda/${{ steps.envname.outputs.name }}.yml
+            conda install --yes --name base pytest
         - name: Run tests
+          shell: bash -l {0}
           run: |
             cd notebooks
+            conda activate base
```

Comment:
I was inclined to agree with you since I included `auto-activate-base: true` in action `conda-incubator`. However, this happened after I removed the statement:

<img width="514" alt="Capture" src="https://user-images.githubusercontent.com/1136006/191314099-45dc3b18-49f4-4cd9-bad2-cb8560e9f4cb.PNG">

so I put it back